scalameta/contrib/shared/src/main/scala/scala/meta/contrib/ScaladocParser.scala

```diff
@@ -43,6 +45,18 @@ object ScaladocParser {
     comment.content.map(parseRec)
   }
 
+  private[this] def generateHeadingParser(level: Int, f: (String) => DocToken): Parser[DocToken] = {
+    val headingSimbols = "=" * level
+    P(
+      // Code block start
+      headingSimbols
+      // Heading description
+        ~ ((AnyChar ~ !"=").rep ~ AnyChar).!.map(f(_))
```

Comment:
I think we can get away with a single header parser if we cut `"=".rep` and inside the map we construct the correct header by the length of the `=`.